### PR TITLE
Like dislike

### PR DESCRIPTION
**Pull request by msdeshmukh009**
### Added like/dislike feature
screens-44

### CODE REVIEW

File: src/components/stacked-video-item/StackedVideoItem.jsx [hunk 1]
```diff
@@ -1,20 +1,27 @@
...
+  const { _id, title } = video;
+
+  const { removeFromLike } = useLikes();
+
+  const getVideoTitleTrimmedToEightyChar = title => {
+    if (title.length < 40) {
+      return title;
+    }
+    return title.substr(0, 38) + "..";
+  };
```
Comment (BelideNikhil) [inline src/components/stacked-video-item/StackedVideoItem.jsx]: you can use ternary here

File: src/pages/authentication/Signin.jsx [hunk 2]
```diff
@@ -1,17 +1,86 @@
...
+
+  const loginWithGuest = e => {
+    e.preventDefault();
+    setSubmitted(true);
+    setUserInput({
+      email: "adarshbalika@gmail.com",
+      password: "adarshBalika123",
+    });
+    login({ email: "adarshbalika@gmail.com", password: "adarshBalika123" }, from);
+  };
```
Comment (BelideNikhil) [inline src/pages/authentication/Signin.jsx]: you can make an object as guest user here

File: src/reducers/likesReducer.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+const { INITIALIZE, SET_ERROR, SET_LIKEDLIST } = likesActions;
+
+const likesReducer = (state, action) => {
+  switch (action.type) {
+    case INITIALIZE:
+      return { ...state, loading: true, error: "" };
+
+    case "SET_LIKEDLIST":
+      return { ...state, likedList: action.payload, loading: false };
+
```
Comment (BelideNikhil) [inline src/reducers/likesReducer.js]: make this a variable as well

File: src/routes/PrivateRoute.jsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  const {
+    authState: {
+      userDetails: { token },
+    },
+  } = useAuth();
+
+  const location = useLocation();
+
+  return token ? <Element /> : <Navigate to="/signin" state={{ from: location }} replace />;
+};
```
Comment (BelideNikhil) [inline src/routes/PrivateRoute.jsx:14]: clever implementation

**Review by BelideNikhil — APPROVED** [review]
GTG!
Nice UI, works flawlessly
The code is nicely divided and readable as well

**Review by tanveertkd — APPROVED** [review]
This is good to go. Good clean code. 👍

**Pull request merged**